ui/src/features/Editor/hooks.ts

```diff
@@ -321,10 +330,53 @@ export default ({
     }
   };
 
+  useEffect(() => {
+    if (yAwareness && !yAwareness.getLocalState()?.color) {
+      const color =
+        CURSOR_COLORS[Math.floor(Math.random() * CURSOR_COLORS.length)];
+      yAwareness.setLocalStateField("color", color);
+      yAwareness.setLocalStateField("clientId", yAwareness.clientID);
+      yAwareness.setLocalStateField("userName", me?.name || "Unknown user");
+    }
+  }, [me, yAwareness]);
```

Comment:
I think this should be done in useYjsSetup before `setYAwareness(yWebSocketProvider.awareness);`. You are doing cleanup of localstate there in the useeffect's return anyways, so it makes sense to do this there too